src/zenml/new/pipelines/pipeline.py

```diff
@@ -1314,17 +1326,9 @@ def with_options(
 
         pipeline_copy = self.copy()
 
-        pipeline_copy._parse_config_file(
-            config_path=config_path,
-            matcher=inspect.getfullargspec(self.configure)[0]
-            + ["model_version"],  # TODO: deprecate `model_version` later on
+        pipeline_copy._reconfigure_from_file_with_overrides(
```

Comment:
I'm still not entirely sure how this whole config file logic works/is supposed to work:
- We apply the config file to the pipeline here
- We also include it in run_args just below, which means that it will get re-applied once `Pipeline._run(...)` is getting called

Maybe what would help me is if you could point out some errors in how I think this behaves:
- I call `pipeline.with_options(config_path="config_a.yaml", enable_cache=...)`: This applies the config file, and overwrites the value with the value for `enable_cache` that was passed in code.
-  I call `pipeline.with_options(config_path="config_b.yaml", enable_cache=...)` again with a different config file. Again, the config file will get applied and the caching config overwritten.
- I now run the pipeline. This applies the config_b file again, which overwrites the value given by `enable_cache` in code in the step before this.

Reply:
TBH, this is also confusing to me. I would keep it clean and forbid overriding the values, so we don't land in such ethical discussions.
As of now we do it for the pipeline parameters, but not for configurations.

I think that what you described is valid, but I never tested anything that complex before.

Reply:
I see. Let's leave it as is in the current PR, but this is something that we should maybe revisit and clean up so we know what exactly is going on at some point 👍